lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitEvaluator.cs

```diff
@@ -0,0 +1,247 @@
+        {
+            private static readonly Logger Logger = Logger.GetLogger(typeof(ResubmitEvaluatorDriver));
+
+            private readonly IEvaluatorRequestor _requestor;
+            private int _taskNumber = 1;
+            private int _contextNumber = 1;
+
+            [Inject]
+            private ResubmitEvaluatorDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(2).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                Logger.Log(Level.Info, "AllocatedEvaluator: " + value.Id);
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, ContextId + _contextNumber)
+                        .Build());
+                Interlocked.Increment(ref _contextNumber);
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                Logger.Log(Level.Info, "ActiveContext: " + value.Id);
+                value.SubmitTask(TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId + _taskNumber)
+                    .Set(TaskConfiguration.Task, GenericType<FailEvaluatorTask>.Class)
+                    .Set(TaskConfiguration.OnMessage, GenericType<FailEvaluatorTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<FailEvaluatorTask>.Class)
+                    .Build());
+                Interlocked.Increment(ref _taskNumber);
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "RunningTask: " + value.Id);
+                string msg = value.Id.Equals(TaskId + "1") ? FailSignal : SuccSignal;
+                value.Send(Encoding.UTF8.GetBytes(msg));
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, CompletedTaskValidationMessage + ". Task completed: " + value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(ICompletedEvaluator value)
+            {
+                Logger.Log(Level.Info, CompletedEvaluatorValidationMessage + ". Evaluator completed: " + value.Id);
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
+                Logger.Log(Level.Info, FailedEvaluatorMessage + ". Evaluator failed: " + value.Id + "  FailedTaskId: " + value.FailedTask.Value.Id);
+                Assert.NotNull(value.FailedTask);
+                Assert.NotNull(value.FailedTask.Value);
+                Assert.NotNull(value.FailedTask.Value.Id);
+                Assert.True(value.FailedTask.Value.Id.Equals(TaskId + "1") || value.FailedTask.Value.Id.Equals(TaskId + "2"));
```

Comment:
I requested two evaluators and two tasks at beginning. Task can be assigned to any one of the evaluators as we don't control the event sequence. It can be TaskId1 or TaskId2, but must be one of them.